bob-common/src/mapper.rs

```diff
@@ -121,6 +121,47 @@ impl Virtual {
         self.vdisks.get(&id).expect("vdisk not found").nodes()
     }
 
+    fn find_support_node_index(
+        nodes: &[Node],
+        target_nodes: &[Node],
+        offset: usize,
+    ) -> SupportIndexResult {
+        let mut avail = 0;
+        for i in 0..nodes.len() {
+            let node = &nodes[i];
+            if node.connection_available() && target_nodes.iter().all(|n| n.index() != node.index())
+            {
+                if avail == offset {
+                    return SupportIndexResult {
+                        index: Some(i),
+                        avail: avail + 1,
+                    };
+                }
+                avail += 1;
+            }
+        }
+        return SupportIndexResult { index: None, avail };
+    }
+
+    fn find_support_node_offset(nodes: &[Node], target_nodes: &[Node], offset: usize) -> usize {
+        let mut res = Self::find_support_node_index(nodes, target_nodes, offset % nodes.len());
```

Comment:
Actually, we know, that we always skip target nodes, so we can calculate local offset as `offset % (nodes.len() - target_nodes.len())`. But to make it safe, at the begging of the function, we should explicitly check that `target_nodes.len() < nodes.len()`. If it is not, then return 0 immediately. Also in `get_support_nodes` the same check can be added as a short-cut: we can return empty vector in this case

Reply:
Done.